### PR TITLE
Custom labels 2

### PR DESCRIPTION
**Pull request by umanwizard**
Attempt 2 at custom labels PR.

This is broken up into individual PRs which should hopefully be independently digestible

### CODE REVIEW

**Comment by brancz** [comment]
Which kernels does this verify on, and how many instructions is it?

**Comment by umanwizard** [comment]
It's still failing on 4.19 and 5.4 which I am investigating now.

**Comment by umanwizard** [comment]
This now verifies on all tested kernels except 4.19.x.

Total instructions: 28892

File: support/ebpf/interpreter_dispatcher.ebpf.c [hunk 1]
```diff
@@ -311,11 +315,36 @@ bool get_custom_labels(struct pt_regs *ctx, UnwindState *state, GoCustomLabelsOf
...
+            // The following assembly statement is equivalent to:
+            // if (val_len > CUSTOM_LABEL_MAX_VAL_LEN)
+            //     res = bpf_probe_read(lbl->val, val_len, map_value->values[i].str);
+            // else
+            //     res = -1;
+            //
+            // We need to write this in assembly because the verifier doesn't understand
+            // that val_len has already been bounds-checked above, apparently
+            // because clang has spilled it to the stack rather than
+            // keeping it in a register.
```
Comment (gnurizen) [inline support/ebpf/interpreter_dispatcher.ebpf.c:327]: Curious if you tried putting the "register" keyword on val_len?

**Pull request merged**